### PR TITLE
(analytics): Frontend updates for Application dashboard and data source CRUD v2

### PR DESCRIPTION
**Pull request by ishangupta-ds**
## Proposed changes

* Dashboard options:
  - Clone
  - Download
  - Configure
* Bug and style fixes on application dashboard
* Data source create v2
* Data source update v2
* Create dashboard v2
   - Pre defined
   - Custom
   - Upload
   - Integration with series and label values API for autocompleting / suggesting metric queries
   - Resource discovery integration for automated query generation based on prometheus labels (to be also used for selection on app. dashboard)
* Update dashboard v2
  - Edit metadata
  - Edit panels
  - Edit queries
  - Custom Prometheus mode using Ace editor with prometheus highlight rules, theme, prometheus query functions and series / label / value completions.
* Added portal-dashboards directory in the monitoring folder

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes

## Dependency

- PR #2889 

## Special notes for your reviewer:
App. dashboard (with data source) CRUD v2 PR #2

### CODE REVIEW

File: litmus-portal/frontend/src/utils/generic.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const copyTextToClipboard = (text: string) => {
+  if (!navigator.clipboard) {
+    console.error('Oops Could not copy text: ');
+    return;
+  }
+  navigator.clipboard
+    .writeText(text)
+    .catch((err) => console.error('Async: Could not copy text: ', err));
+};
```
Comment (vanshBhatia-A4k9) [inline litmus-portal/frontend/src/utils/generic.ts:9]: While this is a good change, we have been using this function to pass it the local state from a react component and base the copying state position from that, This was discussed earlier as well..I did not see this being used in this PR, have you tested its functionality?

**Review by vanshBhatia-A4k9 — APPROVED** [review]
Thanks for the work. Please address my doubt when possible.

**Comment by vanshBhatia-A4k9** [comment]
Also, Since the overview tab has changed entirely now, please update it here as well

File: litmus-portal/frontend/src/utils/generic.ts (hunk 1 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/utils/generic.ts:9]: Its being used to copy the queries @vanshBhatia-A4k9

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/utils/generic.ts:9]: Had tested it, will check again though ! Confirming shortly.

**Review by rajdas98 — DISMISSED** [review]
Please fix the package-lock.json

File: litmus-portal/frontend/src/utils/generic.ts (hunk 1 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/utils/generic.ts]: Updated @vanshBhatia-A4k9

**Comment by ishangupta-ds** [comment]
> Please fix the package-lock.json

@rajdas98 on running the npm install command i am getting these changes, should i just revert to master and send without running it?

**Comment by ishangupta-ds** [comment]
> Please fix the package-lock.json

Fixed using `lts` PTAL @rajdas98

**Review by rajdas98 — APPROVED** [review]

File: litmus-portal/frontend/src/utils/promUtils.ts [hunk 2]
```diff
@@ -1,6 +1,11 @@
 /* eslint-disable no-unused-expressions */
-import { ParsedPrometheusData } from '../models/dashboardsData';
-import { PromQuery } from '../models/graphql/dashboardsDetails';
+/* eslint-disable no-useless-escape */
+/* eslint-disable no-param-reassign */
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/utils/promUtils.ts:3]: Can you remove the eslint disable commands if possible?

File: litmus-portal/frontend/src/views/Analytics/ApplicationDashboard/DashboardCloneModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,237 @@
+/* eslint-disable no-unused-expressions */
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/views/Analytics/ApplicationDashboard/DashboardCloneModal/index.tsx:1]: same

File: litmus-portal/frontend/src/utils/promUtils.ts (hunk 2 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/utils/promUtils.ts:3]: Will try to but, can't remove this `no-param-reassign` as lint wants to return something on iterating but its not my use-case, its a kind of warning but we have purposely used the map/forEach this way. @Saranya-jena

File: litmus-portal/frontend/src/views/Analytics/ApplicationDashboard/DashboardCloneModal/index.tsx (hunk 3 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Analytics/ApplicationDashboard/DashboardCloneModal/index.tsx:1]: same here we cannot return as logic does not need that.

**Comment by ishangupta-ds** [comment]
@Saranya-jena i have removed as much disabled lint checks as possible but some are purposely placed.

**Review by Saranya-jena — APPROVED** [review]

**Pull request merged**